--- nlinum.el	2016-06-18 13:38:43.000000000 +0300
+++ nlinum-patched.el	2016-06-18 13:40:58.000000000 +0300
@@ -81,8 +81,13 @@
             (aref fi 10)
           width)))))
 
+;;; Hack from https://github.com/syl20bnr/spacemacs/issues/6104#issuecomment-220733435
+(defvar frame-ready nil)
+(add-hook 'after-make-frame-functions (lambda (frame) (set-frame-parameter frame 'frame-ready t)))
+(add-hook 'after-init-hook (lambda () (set-frame-parameter nil 'frame-ready t)))
+
 (defun nlinum--setup-window ()
-  (let ((width (if (display-graphic-p)
+  (let ((width (if (and frame-ready (display-graphic-p)) ;; <-- Here
                    (ceiling
                     (let ((width (nlinum--face-width 'linum)))
                       (if width
